File: LeetCode May Challenge/1.week/3-Ransom_Note.cpp
bool canConstruct(string ransomNote, string magazine)
    {
        int ranLen = ransomNote.length();
        int ans = ranLen;
        for(char& c : magazine)
        {
            for(int i = 0; i < ranLen; i++)
            {
                if(c == ransomNote[i])
                {
                    c = '2';
                    ransomNote[i] = '1';
                    ans--;
                    break;
                }
            }
        }
        return !ans;
    }

File: LeetCode May Challenge/1.week/3-Ransom_Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine): 
        def rec(ran,mag):
            if ran == "":
                return True;
            if mag.count(ran[0]) == 0:
                return False
            x = mag.find(ran[0])
            mag = mag[:x] + mag[x+1:]
            return rec(ran[1:],mag)
        return rec(ransomNote,magazine)
